reward/staking_manager.go

```diff
@@ -167,8 +167,14 @@ func CheckStakingInfoStored(blockNum uint64) error {
 	}
 
 	stakingBlockNumber := params.CalcStakingBlockNumber(blockNum)
-	_, err := updateStakingInfo(stakingBlockNumber)
 
+	// skip checking if staking info is stored in DB
+	if stakingInfo, err := getStakingInfoFromDB(stakingBlockNumber); stakingInfo != nil && err == nil {
```

Comment:
Is it possible that stakingInfo is nil and err is nil?

Reply:
No it is not. Just making sure of checking.